components/camel-aws/src/main/java/org/apache/camel/component/aws/sns/SnsEndpoint.java

```diff
@@ -92,6 +107,11 @@ public void doStart() throws Exception {
             LOG.trace("Updating the SNS region with : {} " + configuration.getAmazonSNSEndpoint());
             snsClient.setEndpoint(configuration.getAmazonSNSEndpoint());
         }
+
+        // check the setting the headerFilterStrategy
+        if (headerFilterStrategy == null) {
+            headerFilterStrategy = new SnsHeaderFilterStrategy();
```

Comment:
It might be better to set the the header filter strategy to be Sns from the beginning eg in the top where the instance variable is declared.